### PR TITLE
Some minor fixes to allow dive loading captures of gles application with OpenXR

### PR DESCRIPTION
**Pull request by wangra-google**
* Fix the case where 2 memory blocks have the same gpu address and size but different content
* Hack fix the openxr layer with a dummy xrInitializeLoaderKHR
* Disable the assert of submissions get from perfetto based on vkQueueSubmit (for gl applications)
* Increase the max pending ib to 150
* Fix a warning of uninitialized member var

### CODE REVIEW

File: capture_service/android_application.cc [hunk 1]
```diff
@@ -117,7 +117,8 @@ AndroidApplication::AndroidApplication(AndroidDevice  &dev,
     m_dev(dev),
     m_package(std::move(package)),
     m_type(type),
-    m_started(false)
+    m_started(false),
+    m_is_debuggable(false)
```
Comment (wangra-google) [inline capture_service/android_application.cc:121]: this fix a compiler warning of uninitialized var

File: dive_core/capture_data.cpp [hunk 2]
```diff
@@ -280,7 +280,7 @@ void MemoryManager::Finalize(bool same_submit_copy_only, bool duplicate_ib_captu
                     DIVE_ASSERT(memory_block.m_va_addr == prev_addr);
 
                     // Use whichever one is bigger and get rid of the smaller one
-                    if (memory_block.m_data_size > temp_memory_blocks.back().m_data_size)
+                    if (memory_block.m_data_size >= temp_memory_blocks.back().m_data_size)
```
Comment (wangra-google) [inline dive_core/capture_data.cpp:283]: we have 2 blocks that has the same size and gpu address, but the 2nd one has the most updated content

File: dive_core/common/emulate_pm4.h [hunk 3]
```diff
@@ -214,7 +214,7 @@ class EmulatePM4
         kTotalIbLevels
     };
 
-    static const uint32_t kMaxPendingIbs = 100;
+    static const uint32_t kMaxPendingIbs = 150;
```
Comment (wangra-google) [inline dive_core/common/emulate_pm4.h:217]: it is normal to have more ibs since on a7xx, each tile has it's own ib

**Comment by shanminchao** [comment]
> Some minor fixes to allow dive load captures of systemui

Might want to rename the description. Dive is an open source project and "systemui"
doesn't mean anything in this context.

**Comment by wangra-google** [comment]
> > Some minor fixes to allow dive load captures of systemui
> 
> Might want to rename the description. Dive is an open source project and "systemui" doesn't mean anything in this context.

done

**Comment by shanminchao** [comment]
> > > Some minor fixes to allow dive load captures of systemui
> > 
> > 
> > Might want to rename the description. Dive is an open source project and "systemui" doesn't mean anything in this context.
> 
> done

You might always want to rename the name of the commit itself. Still mentions a "systemui".

Also, why are there 2 commits to this PR?
image

**Comment by wangra-google** [comment]
> > > > Some minor fixes to allow dive load captures of systemui
> > > 
> > > 
> > > Might want to rename the description. Dive is an open source project and "systemui" doesn't mean anything in this context.
> > 
> > 
> > done
> 
> You might always want to rename the name of the commit itself. Still mentions a "systemui".
> 
> Also, why are there 2 commits to this PR? image

Done

The 2nd commit is auto generated when i click "Update branch" (will squash anyways, so doesn't really matter)
image

File: dive_core/common/emulate_pm4.h (hunk 3 above)
Comment (wangra-google) [reply dive_core/common/emulate_pm4.h:217]: randomly chosen :p in this specific case, the max pending is around 130, put 150 just for safety. it is larger than before due to one of my changes to support the CP_START_BIN for a7xx, each bin now has its ib, in this capture there are 66 bins. (Some other PM4s will add pending ibs)

**Comment by shanminchao** [comment]
> > > > > Some minor fixes to allow dive load captures of systemui
> > > > 
> > > > 
> > > > Might want to rename the description. Dive is an open source project and "systemui" doesn't mean anything in this context.
> > > 
> > > 
> > > done
> > 
> > 
> > You might always want to rename the name of the commit itself. Still mentions a "systemui".
> > Also, why are there 2 commits to this PR? image
> 
> Done
> 
> The 2nd commit is auto generated when i click "Update branch" (will squash anyways, so doesn't really matter) image

I still see commit name as [Some minor fixes to allow dive load captures of systemui](https://github.com/google/dive/pull/110/commits/0cc35d159d86ee408ed6390750c406091123c4fd)

I think you changed PR title, not the actual commit name

**Comment by wangra-google** [comment]
> > > > > > Some minor fixes to allow dive load captures of systemui
> > > > > 
> > > > > 
> > > > > Might want to rename the description. Dive is an open source project and "systemui" doesn't mean anything in this context.
> > > > 
> > > > 
> > > > done
> > > 
> > > 
> > > You might always want to rename the name of the commit itself. Still mentions a "systemui".
> > > Also, why are there 2 commits to this PR? image
> > 
> > 
> > Done
> > The 2nd commit is auto generated when i click "Update branch" (will squash anyways, so doesn't really matter) image
> 
> I still see commit name as [Some minor fixes to allow dive load captures of systemui](https://github.com/google/dive/pull/110/commits/0cc35d159d86ee408ed6390750c406091123c4fd)
> 
> I think you changed PR title, not the actual commit name

commit name doesn't really matter after squash, it would be gone

**Comment by shanminchao** [comment]
> > > > > > > Some minor fixes to allow dive load captures of systemui
> > > > > > 
> > > > > > 
> > > > > > Might want to rename the description. Dive is an open source project and "systemui" doesn't mean anything in this context.
> > > > > 
> > > > > 
> > > > > done
> > > > 
> > > > 
> > > > You might always want to rename the name of the commit itself. Still mentions a "systemui".
> > > > Also, why are there 2 commits to this PR? image
> > > 
> > > 
> > > Done
> > > The 2nd commit is auto generated when i click "Update branch" (will squash anyways, so doesn't really matter) image
> > 
> > 
> > I still see commit name as [Some minor fixes to allow dive load captures of systemui](https://github.com/google/dive/pull/110/commits/0cc35d159d86ee408ed6390750c406091123c4fd)
> > I think you changed PR title, not the actual commit name
> 
> commit name doesn't really matter after squash, it would be gone

Ok nevermind! :)

**Pull request merged**